token-lending/program/src/state/reserve.rs

```diff
@@ -95,41 +95,46 @@ impl Reserve {
         obligation: &Obligation,
         liquidity_amount: u64,
         liquidity_token_mint: &Pubkey,
-        token_converter: impl TokenConverter,
+        mut token_converter: impl TokenConverter,
     ) -> Result<LiquidateResult, ProgramError> {
-        // calculate the amount of liquidity that will be repaid
+        // Check obligation health
+        let borrow_token_price = token_converter.best_price(liquidity_token_mint)?;
+        let obligation_collateral_in_liquidity = self
+            .collateral_exchange_rate()?
+            .decimal_collateral_to_liquidity(obligation.deposited_collateral_tokens.into())?;
+        let obligation_loan_to_value = obligation
+            .borrowed_liquidity_wads
+            .try_div(obligation_collateral_in_liquidity)?
+            .try_mul(borrow_token_price)?;
+        let liquidation_threshold = Rate::from_percent(self.config.liquidation_threshold);
+        if obligation_loan_to_value <= liquidation_threshold.into() {
+            return Err(LendingError::HealthyObligation.into());
+        }
+
+        // Calculate the amount of liquidity that will be repaid
         let max_repayable = obligation
             .borrowed_liquidity_wads
             .try_mul(Rate::from_percent(LIQUIDATION_CLOSE_FACTOR))?
             .try_floor_u64()?;
-        let integer_repay_amount = liquidity_amount.min(max_repayable);
-        if integer_repay_amount == 0 {
-            return Err(LendingError::ObligationTooSmall.into());
+
+        // Special handling for dust obligations
+        if max_repayable == 0 {
```

Comment:
Were you able to check that the proptests hit this case?  If not, could you add a special test case for this?

Reply:
Also, can this happen in normal repays too?

Reply:
Yes, there's already handling for normal repays for this case. I'll add some explicit test cases